### PR TITLE
chore(deps): upgrade xstream to 1.4.18

### PR DESCRIPTION
**Pull request by edalex-ian**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [ ] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

Upgrades to latest XStream, while also tidying up some places it's not used (or kind of used - BiRT) and attempting to ensure the same version is used everywhere (even with the Hibernate module) through `val`s in CommonSettings.

Undertook local manual testing of most things. And the first CI run highlighted that the upgrader had issues due to the changes in the security model of XStream - it had changed in 1.4.18 from a black list to a white list for which classes can be safely serialised/deserialised. So we had to explicitly specify the UpgradeLog class.

Thankfully, when Chris Beach did the previous upgrade he did add specifications in the core code for all tle and dytech classes. It's just that doesn't also cover the upgrader.

But will leave the PR in draft while I await a full CI run to pick up things - e.g. like it did with the upgrader/installer.

(Note the move to a new MXParser - they why is in the README here: https://github.com/x-stream/mxparser )

Hopefully resolves #2998

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: Source/Plugins/Birt/org.eclipse.birt.osgi/build.sbt [hunk 1]
```diff
@@ -13,8 +13,7 @@ libraryDependencies := Seq(
   "org.apache.commons"           % "com.springsource.org.apache.commons.httpclient" % "3.1.0",
   "org.apache.commons"           % "com.springsource.org.apache.commons.logging"    % "1.1.1",
   "org.apache.commons"           % "com.springsource.org.apache.commons.codec"      % "1.6.0",
-  "com.thoughtworks.xstream"     % "com.springsource.com.thoughtworks.xstream"      % "1.3.1",
-  "org.xmlpull"                  % "com.springsource.org.xmlpull"                   % "1.1.4.c",
```
Comment (edalex-ian) [inline Source/Plugins/Birt/org.eclipse.birt.osgi/build.sbt:17]: No need to explicitly specify, as it's a transitive of `com.springsource.com.thoughtworks.xstream`.

File: Source/Server/adminTool/build.sbt [hunk 2]
```diff
@@ -4,7 +4,6 @@ libraryDependencies ++= Seq(
   "org.slf4j"              % "jcl-over-slf4j"  % "1.7.33",
   "org.slf4j"              % "slf4j-api"       % "1.7.33",
   "org.slf4j"              % "slf4j-simple"    % "1.7.33",
-  "org.jvnet.hudson"       % "xstream"         % "1.3.1-hudson-8",
```
Comment (edalex-ian) [inline Source/Server/adminTool/build.sbt:7]: Seemed to be unused - everything still built and worked with it removed.

File: autotest/Tests/build.sbt [hunk 3]
```diff
@@ -36,17 +36,16 @@ libraryDependencies ++= Seq(
   "axis"                      % "axis"                     % "1.4",
   "com.jcraft"                % "jsch"                     % "0.1.55",
 //  "jpf" % "jpf-tools" % "1.0.5",
-  "org.jacoco"       % "org.jacoco.report"         % "0.8.7",
-  "org.dspace"       % "oclc-harvester2"           % "0.1.12",
-  "org.jvnet.hudson" % "xstream"                   % "1.3.1-hudson-8",
```
Comment (edalex-ian) [inline autotest/Tests/build.sbt:41]: Seemed only needed in OldTests, and you'll see above I used the official one for that.

**Comment by edalex-ian** [comment]
Nope, the automated install is still have issues with some kind of ForbiddenClassException from XStream.

Will investigate on Monday.

File: Source/Tools/UpgradeInstallation/src/com/tle/upgrade/UpgradeMain.java [hunk 4]
```diff
@@ -170,6 +170,7 @@ public UpgradeMain(File path) {
           "Path '" + tleInstallDir + "' does not appear to be an EQUELLA install directory");
     }
     xstream = new XStream();
+    xstream.allowTypes(new Class[] {com.tle.upgrade.UpgradeLog.class});
```
Comment (edalex-ian) [inline Source/Tools/UpgradeInstallation/src/com/tle/upgrade/UpgradeMain.java:173]: A key change required to support the security model changes in XStream 1.4.18.

File: Source/Plugins/Birt/org.eclipse.birt.osgi/build.sbt (hunk 1 above)
Comment (edalex-ian) [reply Source/Plugins/Birt/org.eclipse.birt.osgi/build.sbt:17]: Well, actually it's even been replaced by their own fork.

**Comment by edalex-ian** [comment]
CI fail is on known flaky: 'LDAPTest'. Should be fine.

**Review by PenghaiZhang — APPROVED** [review]
Thanks Ian!! LGTM

**Pull request merged**